### DISCUSSION TITLE
Check all requests have handlers

### DISCUSSION THREAD

**Discussion by svrooij** [LuckyPennySoftware/MediatR#839] [Q&A] [answered]
Currently the application starts even it there is an IRequest that has no handler.

If I try to execute that request throws an exception that there is no handler in available in dependency injection.

Is it possible to check that each request in the assembly has a handler registered? Either to just throw an error at startup or to run something in a pipeline that that throws an exception.

**Comment by jbogard** [comment]
With an integration test, yes. Look for concrete types that implement `IBaseRequest` and call `Send` for each one of those.

**Comment by StefanoOcchetti** [comment, marked as answer]
May this help?
https://codeopinion.com/find-mediatr-requests-without-handlers/

**Reply by svrooij** [reply]
I actually found this code through my favorite search engine already, but disliked the fact that I had to bring in yet another nuget package. Since I'm using Microsoft.Extensions.DI.

But it looks like a solid solution.
